### PR TITLE
Implement compilation with CMake

### PR DESCRIPTION
**Pull request by sbaldu**
This PR:
* rewrites the compilations of the library modules using CMake
* updates the `setup.py` to use CMake when building the wheel
* checks if Boost is present and if not fetches it (issue #11)
* removes pybind from toml because we use it as a submodule
* include `alpaka` and `pybind11` extern folders in the tarball when packaging

### CODE REVIEW

File: CLUEstering/CLUEstering.py [hunk 1]
```diff
@@ -13,10 +14,26 @@
...
-
-def test_blobs(n_samples: int, n_dim: int, n_blobs: int = 4, mean: float = 0,
+from os.path import dirname, join
+path = dirname(__file__)
+import sys
+sys.path.insert(1, join(path, 'lib'))
+import CLUE_Convolutional_Kernels as clue_kernels
+import CLUE_CPU_Serial as cpu_serial
+tbb_found = False
+if len(glob(join(path, 'lib/CLUE_CPU_TBB*.so'))) != 0:
```
Comment (waredjeb) [inline CLUEstering/CLUEstering.py:24]: Maybe? 
```suggestion
tbb_found = os.path.exists(join(lib_path, 'CLUE_CPU_TBB*.so'))
if(tbb_found):
```

Comment (waredjeb) [inline CLUEstering/CLUEstering.py:24]: Maybe:
```suggestion
tbb_found = os.path.exists(join(lib_path, 'CLUE_CPU_TBB*.so'))
if(tbb_found):
```

And same below?

Comment (waredjeb) [reply CLUEstering/CLUEstering.py:24]: Maybe we should add the `else` statement, telling the user that the backend has not been loaded?

Comment (waredjeb) [reply CLUEstering/CLUEstering.py:24]: Ok perfect :)

Comment (sbaldu) [reply CLUEstering/CLUEstering.py:24]: > Maybe:
> 
> And same below?

Yes we can change it, but we still need to glob, otherwise it doesn't find the path

Comment (sbaldu) [reply CLUEstering/CLUEstering.py]:
```
tbb_found = exists(*glob(join(path, 'lib/CLUE_CPU_TBB*.so')))
```

**Review by waredjeb — APPROVED** [review]

**Comment by waredjeb** [comment]
Failed checks will be fixed in #16

**Pull request merged**